src/libtiled/tilelayer.cpp

```diff
@@ -127,7 +180,10 @@ void Tiled::TileLayer::setCell(int x, int y, const Cell &cell)
         }
     }
 
-    existingCell = cell;
+    chunk(x, y)->setCell(x & CHUNK_MASK, y & CHUNK_MASK, cell);
+
+    if (chunk(x, y)->isEmpty())
```

Comment:
This is the third call to `chunk(x, y)` in this function. You should really call it just once and remember the return value.